index.js

```diff
@@ -1,189 +1,317 @@
- * ```
- *
- * @param {Object} object
- * @param {String} path
- * @return {Object} value or `undefined`
+'use strict';
+
+/* !
+ * Chai - pathval utility
+ * Copyright(c) 2012-2014 Jake Luer <jake@alogicalparadox.com>
+ * @see https://github.com/logicalparadox/filtr
+ * MIT Licensed
  */
 
-exports.get = function(obj, path) {
-  var parsed = exports.parse(path);
-  return getPathValue(parsed, obj);
-};
+var type = require('type-detect');
 
 /**
- * ### .set(path, value, object)
+ * ### .hasProperty(object, name)
  *
- * Define the value in an object at a given string path.
+ * This allows checking whether an object has
+ * named property or numeric array index.
  *
- * ```js
- * var obj = {
- *     prop1: {
+ * Basically does the same thing as the `in`
+ * operator but works properly with natives
+ * and null/undefined values.
+ *
+ *     var obj = {
  *         arr: ['a', 'b', 'c']
  *       , str: 'Hello'
  *     }
- *   , prop2: {
- *         arr: [ { nested: 'Universe' } ]
- *       , str: 'Hello again!'
- *     }
- * };
- * ```
  *
- * The following would be acceptable.
+ * The following would be the results.
  *
- * ```js
- * var properties = require('tea-properties');
- * properties.set(obj, 'prop1.str', 'Hello Universe!');
- * properties.set(obj, 'prop1.arr[2]', 'B');
- * properties.set(obj, 'prop2.arr[0].nested.value', { hello: 'universe' });
- * ```
+ *     hasProperty('str', obj);  // true
+ *     hasProperty('constructor', obj);  // true
+ *     hasProperty('bar', obj);  // false
+ *
+ *     hasProperty('length', obj.str); // true
+ *     hasProperty(1, obj.str);  // true
+ *     hasProperty(5, obj.str);  // false
+ *
+ *     hasProperty('length', obj.arr);  // true
+ *     hasProperty(2, obj.arr);  // true
+ *     hasProperty(3, obj.arr);  // false
  *
  * @param {Object} object
- * @param {String} path
- * @param {Mixed} value
+ * @param {String|Number} name
+ * @returns {Boolean} whether it exists
+ * @namespace Utils
+ * @name hasProperty
  * @api public
  */
 
-exports.set = function(obj, path, val) {
-  var parsed = exports.parse(path);
-  setPathValue(parsed, val, obj);
+var literals = {
+  'number': Number,
+  'string': String,
 };
+function hasProperty(name, obj) {
+  var objType = type(obj);
+  // Bad Object, obviously no props at all
+  if (objType === 'null' || objType === 'undefined') {
+    return false;
+  }
 
-/*!
+  // The `in` operator does not work with certain literals
+  // box these before the check
+  if (literals[objType] && typeof obj !== 'object') {
+    obj = new literals[objType](obj);
+  }
+
+  return name in obj;
+}
+
+/* !
+ * ## parsePath(path)
+ *
  * Helper function used to parse string object
- * paths. Use in conjunction with `getPathValue`.
+ * paths. Use in conjunction with `internalGetPathValue`.
  *
- *  var parsed = parsePath('myobject.property.subprop');
+ *      var parsed = parsePath('myobject.property.subprop');
  *
  * ### Paths:
  *
  * * Can be as near infinitely deep and nested
```

Comment:
Yes, makes sense.
I also noticed that I forgot to update some other strings here I have changed before but reseted my branch.
I'll push changes in minutes.